From 5f4813c5d4ce7496674547fc8d15b84d541448ae Mon Sep 17 00:00:00 2001
From: Nicola Guerrera <guerrera.nicola@gmail.com>
Date: Tue, 5 Aug 2025 18:38:27 +0200
Subject: [PATCH] 2/2] Add pm-wakeup nodes backport from Linux-6.16[

---
 arch/arm64/configs/sm8150.config | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/include/linux/pm_wakeup.h b/include/linux/pm_wakeup.h
index e4e6fb838f81..4be88628474f 100644
--- a/include/linux/pm_wakeup.h
+++ b/include/linux/pm_wakeup.h
@@ -241,3 +241,20 @@ static inline int device_init_wakeup(struct device *dev, bool enable)
 }
 
+static void device_disable_wakeup(void *dev)
+{
+	device_init_wakeup(dev, false);
+}
+
+/**
+* devm_device_init_wakeup - Resource managed device wakeup initialization.
+* @dev: Device to handle.
+*
+* This function is the devm managed version of device_init_wakeup(dev, true).
+*/
+static inline int devm_device_init_wakeup(struct device *dev)
+{
+	device_init_wakeup(dev, true);
+	return devm_add_action_or_reset(dev, device_disable_wakeup, dev);
+}
+
 #endif /* _LINUX_PM_WAKEUP_H */
-- 
2.51.0

